qiskit/transpiler/passes/routing/swap_strategies/commuting_2q_block.py

```diff
@@ -0,0 +1,57 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A gate made of commuting two-qubit gates."""
+
+from typing import Set
+
+from qiskit import QiskitError
+from qiskit.circuit import Gate
+from qiskit.dagcircuit import DAGOpNode
+
+
+class Commuting2qBlocks(Gate):
+    """A gate made of commuting two-qubit gates.
+
+    This gate is intended for use with commuting swap strategies to make it convenient
+    for the swap strategy router to identify which blocks of operations commute.
+    """
+
+    def __init__(self, node_block: Set[DAGOpNode]):
+        """Initialize the op.
+
+        Args:
+            node_block: A block of nodes that commute.
+
+        Raises:
+            QiskitError: If the nodes in the node block do not apply to two-qubits.
+        """
+        qubits, cbits = set(), set()
+        for node in node_block:
+            if len(node.qargs) != 2:
+                raise QiskitError(f"Node {node.name} does not apply to two-qubits.")
+
+            qubits.update(node.qargs)
+            cbits.update(node.cargs)
+
+        if cbits:
+            raise QiskitError(
+                f"{self.__class__.__name__} does not accept nodes with classical bits."
+            )
+
+        super().__init__("Commuting 2Q gates", num_qubits=len(qubits), params=[])
```

Comment:
The `name` is typically reflecting the gate class name and `snake_case` whereas the `label` can be what you want to be displayed in drawings
```suggestion
        super().__init__("commuting_2q_block", num_qubits=len(qubits), params=[], label="Commuting 2q gates")
```